CMakeLists.txt

```diff
@@ -134,9 +136,7 @@ add_subdirectory(ode-essentials)
 add_subdirectory(ode-rasterizer)
 add_subdirectory(ode-logic)
 if(ODE_RENDERER_ENABLED)
-    if(NOT EMSCRIPTEN)
-        add_subdirectory(ode-media)
-    endif()
+    add_subdirectory(ode-media)
```

Comment:
I added `ode-media` to Emscripten build so that the functionality that works with images in memory (either loading an encoded binary representation of an image to a bitmap or encoding a bitmap to an image) could be used in `ode-renderer`, namely the `loadImage` and `savePng` functions used in `ode_loadDesignFromFileWithImages` and `ode_saveDesignToFileWithImages` -  https://github.com/opendesigndev/open-design-engine/pull/63/files#diff-da1264d3bb6785c89fbcf44b56a1d50a8827c7d23a060d805176204d5be6c553R181-R220 .

I suppose an alternative would be not to load/save images from/to binary octopus files in Emscripten build.

Reply:
Ok, that makes sense, but in that case, image format libraries need to be linked to the Emscripten build, I am not sure if you took care of that. However, this is something that will probably make the WASM package much bigger which is something we want to avoid, so at the very least, I would make this configurable in CMake.

Reply:
Also, Emscripten does not use the filesystem and it seems to me that you only implemented loading / saving Octopus files in the filesystem, not in memory, so adding Media to Emscripten doesn't seem necessary at least at this stage.

Reply:
> Also, Emscripten does not use the filesystem and it seems to me that you only implemented loading / saving Octopus files in the filesystem, not in memory, so adding Media to Emscripten doesn't seem necessary at least at this stage.

Ok. I guess in that case disabling the api functions that load and save octopus files for Emscripten would be a solution.

Reply:
Fixed.